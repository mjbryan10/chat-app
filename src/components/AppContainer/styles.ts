import styled from 'styled-components';
import { breakpoint } from 'shared/styles/Variables';

interface DisplaySwitchProps {
   active: boolean;
}

export const Container = styled.main`
   display: flex;
   flex-flow: row nowrap;
   max-width: 100vw;
   min-height: 80vh;
`;

export const DisplaySwitch = styled.div<DisplaySwitchProps>`
   display: ${({ active }) => (active ? 'block' : 'none')};
   width: 95%;
   margin: 1em auto;
   @media ${breakpoint.tablet} {
      margin: 1rem;
      display: block;
      width: auto;
      &:last-of-type {
         flex-grow: 1;
         margin-left: 0;
      }
   }
`;
