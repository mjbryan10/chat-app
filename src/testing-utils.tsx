import React from 'react';
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { dark } from './shared/theme';

const AllTheProviders = ({ children }: any) => {
  return (
    <ThemeProvider theme={dark}>
        {children}
    </ThemeProvider>
  )
  
}

const customRender = (ui?: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }