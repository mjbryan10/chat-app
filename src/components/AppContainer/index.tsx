import React from 'react';
import * as S from './styles';
import MessagesContainer from 'features/message/MessagesContainer';
import ConversationContainer from 'features/conversation/ConversationContainer';

const AppContainer = () => {
  return (
    <S.Container>
      <ConversationContainer />
      <MessagesContainer />
    </S.Container>
  )
}

export default AppContainer
