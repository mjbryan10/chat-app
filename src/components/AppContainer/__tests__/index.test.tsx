import React from 'react';
import { render } from 'testing-utils';
import AppContainer from '..';

/**
 * scrollIntoView is not a function fix.
 * @see https://stackoverflow.com/a/60225417/12873927
 */
window.HTMLElement.prototype.scrollIntoView = jest.fn()

describe('AppContainer Component', () => {
   test('should render correctly', () => {
     const { asFragment } = render(<AppContainer />);

      expect(asFragment()).toMatchSnapshot();
   });
});
