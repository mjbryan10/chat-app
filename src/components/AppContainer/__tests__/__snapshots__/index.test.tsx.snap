// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer Component should render correctly 1`] = `
<DocumentFragment>
  <main
    class="sc-AxjAm eeUVjI"
  >
    <div
      class="sc-AxirZ gFsXol"
    >
      <div
        class="sc-fzokOt hwomJI"
      >
        <h2
          class="sc-fzqBZW kFYROT"
        >
          Conversations
        </h2>
        <div
          class="sc-fzoXzr hCDWur"
        />
        <div
          class="sc-fzoyAV joLyXM"
        >
          <button
            class="sc-fzqNJr dVPdEh"
          >
            <svg
              fill="currentColor"
              height="3rem"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="3rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
              />
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
              />
            </svg>
            Create new conversation
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-AxirZ hAsYYO"
    >
      <div
        class="sc-fzplWN cUeYUu"
      >
        <div
          class="sc-fznyAO hCupfQ"
        >
          <div />
        </div>
        <form
          class="sc-fzpans bpFApF"
          data-testid="message-creator-form"
        >
          <textarea
            class="sc-fzoLsD hnRPQP"
            data-testid="message-creator-textbox"
            disabled=""
            placeholder="Select a conversation"
          />
          <input
            class="sc-fzozJi PbjRk"
            data-testid="input-submit-button"
            type="submit"
            value="Send"
          />
        </form>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
